### ISSUE TITLE
Compiler error:"mismatched types `string` and `&string`"???

### ISSUE THREAD

**Issue by donaldcallen** [vlang/v#6274]
**V version:** 
V 0.1.29 cbcba2e
**OS:**
64-bit Arch Linux, up-to-date 

**What did you do?**
```v
type Foo = int | string
fn main () {
	mut bar := Foo(5)
	match bar {
		int {println("it's an int: $bar")}
		string {println("shouldn't happen")}
	}
	bar = Foo('hello')
	match bar {
		string {println("it's a string: "+bar)}
		int {println("shouldn't happen")}
	}
}
```

**What did you expect to see?**
An executable that, when run, prints
```
It's an int: 5
It's a string: hello
```
 
**What did you see instead?**
The following from the compiler:
```
dca@franz:~/Software/v_issues$ ../v/v main.v
main.v:10:42: error: mismatched types `string` and `string`
    8 |     bar = Foo('hello')
    9 |     match bar {
   10 |         string {println("it's a string: "+bar)}
      |                                          ^
   11 |         int {println("shouldn't happen")}
   12 |         }
dca@franz:~/Software/v_issues$
```

**Comment by danieldaeschle** [comment]
Post your code please.

**Comment by donaldcallen** [comment]
Read my original post more carefully, please.

**Comment by Delta456** [comment]
This works. It first references and then deferences it but it shouldn't be the case

```v
type Foo = int | string
fn main () {
	mut bar := Foo(5)
	match bar {
		int {println("it's an int: $bar")}
		string {println("shouldn't happen")}
	}
    
	bar = Foo('hello')
	match bar {
		string {println("it's a string: "+ *(&bar))}
		int {println("shouldn't happen")}
	}
    
}
```

**Comment by spytheman** [comment]
Fixed by #6342

**Comment by Larpon** [comment]
Please re-open if the issue is still valid 🙂
